subsys/net/ip/net_pkt.c

```diff
@@ -1873,6 +1873,7 @@ int net_pkt_pull(struct net_pkt *pkt, size_t length)
 {
 	struct net_pkt_cursor *c_op = &pkt->cursor;
 	struct net_pkt_cursor backup;
+	bool restore = true;
 
 	if (!pkt->buffer) {
 		return -ENOBUFS;
```

Comment:
btw you added that check in the previous patch, but that's superfluous since if the packet has no buffer the cursor should not point to anything (if it does, it's a bug induced by the caller of this function.), it should be NULL. Thus why there is a test at line 1889.